pandas/core/groupby/groupby.py

```diff
@@ -689,7 +689,19 @@ def _set_result_index_ordered(self, result):
             result.set_axis(index, axis=self.axis, inplace=True)
             result = result.sort_index(axis=self.axis)
 
-        result.set_axis(self.obj._get_axis(self.axis), axis=self.axis, inplace=True)
+        # result.index is a standard index => may need to restore original index
+
+        if (  # check if rows were dropped
```

Comment:
this should call / use _set_result_index_ordered e.g. either use it or something like it. this is too much code in-line and very hard to understand.

Reply:
separated out this part into `_reset_grouper_index` helper method (immediately below)